lib/multi.c

```diff
@@ -2625,8 +2625,12 @@ CURLMcode curl_multi_setopt(struct Curl_multi *multi,
     break;
   case CURLMOPT_PIPELINING_SERVER_BL:
     break;
-  case CURLMOPT_MAX_HTTP2_CONCURRENT_STREAMS:
-    multi->max_http2_concurrent_streams = va_arg(param, uint32_t);
+  case CURLMOPT_MAX_CONCURRENT_STREAMS:
+    {
+      long streams = va_arg(param, long);
+      multi->max_concurrent_streams = (streams>INITIAL_MAX_CONCURRENT_STREAMS)?
+          INITIAL_MAX_CONCURRENT_STREAMS : streams;
```

Comment:
This will allow negative values to be set. I presume that isn't intended? And zero.

Reply:
Agree, updated